### PR TITLE
WT-12397 Add collection and index specific counting.

### PR DESCRIPTION
**Pull request by sueloverso**

### CODE REVIEW

**Comment by sueloverso** [comment]
@ddanderson this PR is still in Draft mode. It works but it doesn't feel very clean. I want to collect a lot of global statistics across all the files. Do you have any suggestions?

File: tools/backup_analysis.py [hunk 1]
```diff
@@ -30,6 +30,76 @@
...
+
+assert(len(global_names) == len(global_types))
+
+blocks_global = dict()
+bytes_global = dict()
+files_changed = dict()
+files_global = dict()
+gran_blocks_global = dict()
+pct20_global = dict()
+pct80_global = dict()
```
Comment (ddanderson) [inline tools/backup_analysis.py:51]: I think what may help is a class, like:
```
class TypeStats(object):
   def __init__(self, name):
      self.name = name
      self.blocks = 0
      self.bytes = 0
      self.files = 0
      ...etc...
```
To get it initialized,
```
typestats = dict()
typestats['coll'] = TypeStats('Replicated Collections')
typestats['index'] = TypeStats('Replicated Indexes')
...
```
Later, you can do:
```
      ts = typestats[count_type]
      ....
      ts.bytes += partial_cmp
      ....
      ts.pct20 += 1

**Review by ddanderson — COMMENTED** [review]
@sueloverso, perhaps a different way to organize the stats you collect.  Not sure `TypeStats` is a good name.

File: tools/backup_analysis.py (hunk 1 above)
Comment (sueloverso) [reply tools/backup_analysis.py:51]: Thank you! That is so much cleaner.

File: tools/backup_analysis.py [hunk 2]
```diff
@@ -53,18 +138,26 @@ def check_backup(mydir):
...
     f1_size = os.stat(os.path.join(dir1, filename)).st_size
     f2_size = os.stat(os.path.join(dir2, filename)).st_size
     min_size = min(f1_size, f2_size)
+    #
+    # Figure out what kind of table this file is. Use the first directory for basis.
+    #
+    metadata = get_metadata(dir1, filename)
+    assert(metadata != None)
+    count_type = compute_type(filename, metadata)
+    ts = typestats[count_type]
```
Comment (ddanderson) [inline tools/backup_analysis.py:152]: Maybe a good sanity check is to check the other directory and make sure the type matches.

File: tools/backup_analysis.py [hunk 3]
```diff
@@ -127,38 +223,115 @@ def compare_file(dir1, dir2, filename, cmp_size, granularity):
...
+        total_bytes += ts.bytes
+        total_files += ts.files
+        total_files_changed += ts.files_changed
+        total_gran_blocks += ts.gran_blocks
+    print(f'Total: {total_bytes} bytes changed in {total_gran_blocks} granularity-sized ({granularity}) blocks')
+    if total_files_changed == 1:
+        word = 'file'
+    else:
+        word = 'files'
+    print(f'Total: {total_files_changed} {word} changed out of {total_files} total files')
```
Comment (ddanderson) [inline tools/backup_analysis.py:272]: Maybe cleaner with an extra function:
```
def plural(s, count):
    if count == 1:
       return s
    else
       return s + 's'

print(f'.....out of {total_files} total {plural("file", total_files_changed)}')
```
If that's too busy, you can use the function along with the `word` variable.

**Review by ddanderson — APPROVED** [review]
Looks good - tagged some suggestions for improvement.

File: tools/backup_analysis.py (hunk 3 above)
Comment (sueloverso) [reply tools/backup_analysis.py:272]: I like something like that. I'll make some similar sort of change and see if it improves things. The repeated conditionals is not ideal.

File: tools/backup_analysis.py (hunk 2 above)
Comment (sueloverso) [reply tools/backup_analysis.py:152]: I like that idea. I will just incorporate it into the compute_type function.

**Comment by sueloverso** [comment]
@ddanderson  I made a bunch of changes based on your suggestions as well as adding a bunch of comments, etc. Please give it one more look.

**Review by ddanderson — APPROVED** [review]
LGTM!

**Pull request merged**